packages/edit-site/src/components/global-styles/screen-revisions/index.js

```diff
@@ -55,13 +55,37 @@ function ScreenRevisions() {
 	const { setEditorCanvasContainerView } = unlock(
 		useDispatch( editSiteStore )
 	);
+	const selectedRevisionMatchesEditorStyles = areGlobalStyleConfigsEqual(
+		globalStylesRevision,
+		userConfig
+	);
+	const shouldSelectFirstItem =
+		! selectedRevisionMatchesEditorStyles &&
+		( ! selectedRevisionId || 'unsaved' === selectedRevisionId );
 
 	useEffect( () => {
 		if ( editorCanvasContainerView !== 'global-styles-revisions' ) {
 			goBack();
 			setEditorCanvasContainerView( editorCanvasContainerView );
+			return;
+		}
+
+		/*
+			Ensure that the first item is selected where no revision is selected
+			and the selected styles don't match the current editor styles.
+			This is required in case editor styles are changed outside the revisions panel,
+			e.g., via the reset styles function of useGlobalStylesReset().
+			See: https://github.com/WordPress/gutenberg/issues/55866
+		*/
+		if ( shouldSelectFirstItem ) {
+			setGlobalStylesRevision( revisions[ 0 ] );
 		}
-	}, [ editorCanvasContainerView ] );
+	}, [
+		editorCanvasContainerView,
+		shouldSelectFirstItem,
+		revisions[ 0 ],
```

Comment:
I think using `revisions` seems to cause the exhaustive-deps rule to expect `revisions` in the dependency array somewhere. If we only need the first element, I wonder if we can grab the first element outside of the useEffect via `const firstRevision = revisions[ 0 ]` and use that? Just a thought!

Reply:
This could work, thanks!

I'll try it.